processors/ratelimiter/ratelimiterprocessor.go

```diff
@@ -0,0 +1,95 @@
+package ratelimiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pb_struct "github.com/envoyproxy/go-control-plane/envoy/extensions/common/ratelimit/v3"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/metadata"
+)
+
+type processor struct {
+	rateLimitServiceClient   pb.RateLimitServiceClient
+	domain                   string
+	domainSoftLimitThreshold uint32
+	logger                   *zap.Logger
+	tenantIDHeaderName       string
+}
+
+const (
+	TenantID       = "tenant_id"
+	GlobalTenantID = "global_tenant_id"
+)
+
+// ProcessTraces implements processorhelper.ProcessTracesFunc
+func (p *processor) ProcessTraces(ctx context.Context, traces ptrace.Traces) (ptrace.Traces, error) {
+	tenantId, err := p.getTenantId(ctx)
+	if err != nil {
+		return traces, nil
+	}
+	// two descriptors, one for tenant and other one for domain
+	desc := make([]*pb_struct.RateLimitDescriptor, 2)
+	desc[0] = &pb_struct.RateLimitDescriptor{
+		Entries: []*pb_struct.RateLimitDescriptor_Entry{
+			{
+				Key:   TenantID,
+				Value: tenantId,
+			},
+		},
+	}
+	desc[1] = &pb_struct.RateLimitDescriptor{
+		Entries: []*pb_struct.RateLimitDescriptor_Entry{
+			{
+				Key:   TenantID,
+				Value: GlobalTenantID,
+			},
+		},
+	}
+	spanCount := uint32(traces.SpanCount())
+	response, err := p.rateLimitServiceClient.ShouldRateLimit(
+		ctx,
+		&pb.RateLimitRequest{
+			Domain:      p.domain,
+			Descriptors: desc,
+			HitsAddend:  spanCount,
```

Comment:
HitsAdded?

Reply:
It's count which needs to be incremented for descriptors.

Reply:
Oh I was referring to the typo.

Reply:
Ignore this. Just realized it's in the api.